### PR TITLE
Feat/total web requests

### PR DESCRIPTION
**Pull request by jgryffindor**
## Description

- Adding a new endpoint for different transform functions to do math and data transforms against metrics stored in the local DB
- In this first case we needed a way to sum the values of the series outside of Prometheus to get totals and a series of increasing values.
- Adding conditionals and new transform prop to metricChart and metricStat components to set different transforms
- Adding new client api functions to dynamically retrieve current/series values from different transform endpoints
- Added an enable column and feature to enable/disable Prometheus queries entity and the metrics scheduler. This allows us to enable/disable collection on stored queries but leave their stored values intact. 

## Testing
- Tests were run locally to verify the endpoint functionality and metric chart rendering

## Screenshots (if applicable)
image

## Checklist:

- [x] I have read and followed the CONTRIBUTING guidelines for this project.
- [ ] I have added or updated tests and they pass.
- [ ] I have added or updated documentation and it is accurate.
- [x] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: server/src/metrics/transform/transforms.service.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    to: Date,
+  ): Promise<SeriesEntity[] | null> {
+    const prometheusQuery = await this.prometheusQuery.get(name);
+
+    const query = this.metricRepository
+      .createQueryBuilder("m")
+      .where("m.prometheusQueryId = :prometheusQuery", {
+        prometheusQuery: prometheusQuery.id,
+      })
+      .orderBy("m.timestamp", "DESC");
```
Comment (fmorency) [inline server/src/metrics/transform/transforms.service.ts:67]: Add a comment that this query returns *every data point* for that given metric, being a year or 10.

File: server/src/metrics/transform/transforms.service.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  // Get the current value for a metric
+  async getSumTotal(name: string): Promise<Current> {
+    const prometheusQuery = await this.prometheusQuery.get(name);
+
+    const query = this.metricRepository
+      .createQueryBuilder("m")
+      .where("m.prometheusQueryId = :prometheusQuery", {
+        prometheusQuery: prometheusQuery.id,
+      })
+      .orderBy("m.timestamp", "DESC");
```
Comment (fmorency) [inline server/src/metrics/transform/transforms.service.ts:33]: Add a comment that this query returns every data point for that given metric, being a year or 10.

**Review by fmorency — CHANGES_REQUESTED** [review]
Thanks!
Some questions/comments.

nit: We'll need to be efficient in the future; returning all data points and computing the sum on-the-fly every time is wrong.

File: server/src/metrics/transform/transforms.service.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      data.push(Number(series.data));
+      timestamps.push(series.timestamp);
+    });
+
+    // Reformat the data to be a sum of previous values at each timestamp
+    for (let i = data.length - 1; i >= 0; i--) {
+      if (i < data.length - 1) {
+        data[i] = data[i] + data[i + 1];
+      }
+    }
```
[the remark this answers is not in the document]
Comment (jgryffindor) [reply server/src/metrics/transform/transforms.service.ts]: Thanks for the tip, reduceRight is much cleaner.

**Review by fmorency — APPROVED** [review]
Thanks!

Please create an issue regarding what we discussed this morning and link it to this PR.

**Pull request merged**